### PR TITLE
 [Core] enable lora for sdxl adapters too and add slow tests.

### PR DESCRIPTION
**Pull request by ilisparrow**
Fixes https://github.com/huggingface/diffusers/issues/5516 
Additionally, adds SLOW tests for SDXL T2i adapters.

Hello, 
as requested here is a PR to fix the #5516, WIth it's corresponding tests.

I have a few questions : 
- The test seem to fail when this is removed from the pipeline initialization : `torch_dtype=torch.float16`,  But your PR that inspired mine didn't have it(https://github.com/huggingface/diffusers/pull/4666/files).
  This is the error that I get with it : 
`  RuntimeError: Input type (float) and bias type (c10::Half) should be the same`
- I get this warning when I add `.to("CPU")` to the pipeline and the adapter : 
`Pipelines loaded with 'torch_dtype=torch.float16' cannot run with 'cpu' device. It is not recommended to move them to 'cpu' as running them will fail. Please make sure to use an accelerator to run the pipeline in inference, due to the lack of support for'float16' operations on this device in PyTorch. Please, remove the 'torch_dtype=torch.float16' argument, or use another device for inference.`
- Last, In the tests, we load this image : https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/t2i_adapter/toy_canny.png, but if we are testing for canny of functions that process images with canny, shouldn't we input a real image and expect this as an output ? Because otherwise we are applying a canny to an already canny image.

I hope this helps. 

@sayakpaul

### CODE REVIEW

**Comment by sayakpaul** [comment]
LoRA related additions look good to me! Are there any cool results that you would like so share with us here? 

For the tests, I will defer to @DN6.

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py [hunk 1]
```diff
@@ -560,3 +564,64 @@ def test_inference_batch_single_identical(
...
+                    "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/t2i_adapter/toy_canny.png"
+        )
+
+        images = pipe(prompt, image=image, generator=generator, output_type="np", num_inference_steps=3).images
+
+        assert images[0].shape == (768, 512, 3)
+
+        original_image = images[0, -3:, -3:, -1].flatten()
+        expected_image = np.array([0.50346327, 0.50708383, 0.50719553, 0.5135172,  0.5155377,  0.5066059, 0.49680984, 0.5005894,  0.48509413])
+        assert np.allclose(original_image, expected_image, atol=1e-04)
```
Comment (DN6) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py:626]: Use `numpy_cosine_similarity_distance` on the arrays

```suggestion
        assert numpy_cosine_similarity_distance(original_image, expected_image) < 1e-4
```

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py [hunk 2]
```diff
@@ -560,3 +564,64 @@ def test_inference_batch_single_identical(
...
+        image = load_image(
+                    "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/t2i_adapter/toy_canny.png"
+        )
+
+        images = pipe(prompt, image=image, generator=generator, output_type="np", num_inference_steps=3).images
+
+        assert images[0].shape == (768, 512, 3)
+
+        original_image = images[0, -3:, -3:, -1].flatten()
+        expected_image = np.array([0.5226907,  0.5272584,  0.52255356, 0.50762856, 0.50497484, 0.50656825, 0.5065469,  0.51809096, 0.48973307])
```
Comment (DN6) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py:599]: Use `numpy_cosine_similarity_distance` on the arrays

```suggestion
        assert numpy_cosine_similarity_distance(original_image, expected_image) < 1e-4
```

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Comment by ilisparrow** [comment]
Thanks, great. Do you know when it will be merged?
I have some feedback or comments in or comment, that you might find interesting.

**Comment by DN6** [comment]
@ilisparrow I think you would need to run `make style && make quality` as well as `make fix-copies` on the PR again.

**Comment by patrickvonplaten** [comment]
Great job @ilisparrow !

**Comment by patrickvonplaten** [comment]
Actually this PR was very much incorrect, SDXL Adapter already had the LoRA load mixin and the tests had errors. Reverting this PR here: https://github.com/huggingface/diffusers/pull/5555

**Comment by ilisparrow** [comment]
Hello, 
Thank you for taking the time to have a look. 
This PR solves this problem : https://github.com/huggingface/diffusers/issues/5516
That [sayakpaul](https://github.com/sayakpaul) was able to reproduce. Where I tried to add alora to an adapter constrained SDXL model. 
It was inspired by this one : https://github.com/huggingface/diffusers/issues/4664 
Which was solving the same for controlnet and SDXL .

I might be missing something, but it fixed a real problem, and was  reproduced by Sayakpaul.

**Comment by sayakpaul** [comment]
That might have been the case because I reproduced it with the stable release of diffusers. Installing from main might have actually solved it:

https://github.com/huggingface/diffusers/blob/75ea54a1512ac443d517ab35cb9bf45f8d6f326e/src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_xl_adapter.py#L127

I apologize for the oversight caused on my part.

**Pull request merged**